src/main/java/com/algorand/algosdk/v2/client/algod/TealDisassemble.java

```diff
@@ -0,0 +1,68 @@
+package com.algorand.algosdk.v2.client.algod;
+
+import com.algorand.algosdk.v2.client.common.Client;
+import com.algorand.algosdk.v2.client.common.HttpMethod;
+import com.algorand.algosdk.v2.client.common.Query;
+import com.algorand.algosdk.v2.client.common.QueryData;
+import com.algorand.algosdk.v2.client.common.Response;
+import com.algorand.algosdk.v2.client.model.DisassembleResponse;
+
+
+/**
+ * Given the base64 encoded program bytes, return the TEAL source code in plain
+ * text. This endpoint is only enabled when a node's configuration file sets
+ * EnableDeveloperAPI to true.
+ * /v2/teal/disassemble
+ */
+public class TealDisassemble extends Query {
+
+    public TealDisassemble(Client client) {
+        super(client, new HttpMethod("post"));
+    }
+
+    /**
+     * TEAL program binary to be disassembled
+     */
+    public TealDisassemble source(byte[] source) {
```

Comment:
Unlike the Go SDK this one seemed to pick up in the byte array input properly. But the base64 comment is still wrong on this class. For consistency I'd recommend getting rid of the TEAL disassembly changes to all SDKs, since it's a bit odd if only some of them support the feature.